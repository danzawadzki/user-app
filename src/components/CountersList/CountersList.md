Default CountersList component with all props:

```jsx
const counter = {
	number: 997,
	label: 'Police'
};
<CountersList counters={[counter, counter, counter]} />;
```

CountersList component with out props:

```jsx
<CountersList />
```
