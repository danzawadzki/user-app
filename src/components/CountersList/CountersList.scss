@import '../../assets/styles/colors';

.CountersList {
	list-style: none;
	padding: 0;
	margin: 0;
	display: flex;

	&__item {
		padding: 0 3rem;
		border-right: 1px solid $app__borders;

		&:first-of-type {
			padding-left: 0;
		}

		&:last-of-type {
			padding-right: 0;
			border: none;
		}
	}
}
