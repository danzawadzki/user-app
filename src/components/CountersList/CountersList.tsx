import * as React from 'react';
import './CountersList.scss';
import Counter, { ICounter } from '../Counter/Counter';

export interface ICountersList {
	/** Array of objects representing counters */
	counters?: ICounter[];
}

/**
 * Counters list component.
 *
 * @version 1.0.0
 */
const CountersList = ({ counters = [] }: ICountersList) => {
	if (counters.length === 0) {
		return <p className="CountersList__message">No counters...</p>;
	}
	return (
		<ul className="CountersList">
			{counters.map(({ number, label }: ICounter, index) => (
				<li className="CountersList__item" key={`counter-${index}`}>
					<Counter number={number} label={label} />
				</li>
			))}
		</ul>
	);
};

export default CountersList;
