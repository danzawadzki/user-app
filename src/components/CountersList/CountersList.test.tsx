import React from 'react';
import { shallow } from 'enzyme';
import CountersList from './CountersList';

describe('CountersList component: ', () => {
	//Shallow rendered component
	const counter = {
		number: 997,
		label: 'Police'
	};
	const counters = [counter, counter, counter];

	const component = shallow(<CountersList counters={counters} />);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should have length of counters array', () => {
		expect(component.find('li')).toHaveLength(counters.length);
	});

	it('should render message "No counters...", if no counters were provided', () => {
		const componentEmpty = shallow(<CountersList />);

		expect(componentEmpty.html()).toContain('No counters...');
	});
});
