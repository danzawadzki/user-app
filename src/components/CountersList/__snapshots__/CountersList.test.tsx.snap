// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountersList component:  should render match snapshot 1`] = `
<ul
  className="CountersList"
>
  <li
    className="CountersList__item"
    key="counter-0"
  >
    <Counter
      label="Police"
      number={997}
    />
  </li>
  <li
    className="CountersList__item"
    key="counter-1"
  >
    <Counter
      label="Police"
      number={997}
    />
  </li>
  <li
    className="CountersList__item"
    key="counter-2"
  >
    <Counter
      label="Police"
      number={997}
    />
  </li>
</ul>
`;
